frame/transaction-payment/asset-tx-payment/src/lib.rs

```diff
@@ -282,18 +288,35 @@ where
 					let actual_fee = pallet_transaction_payment::Pallet::<T>::compute_actual_fee(
 						len as u32, info, post_info, tip,
 					);
+
+					let min_converted_fee =
+						if actual_fee.is_zero() { Zero::zero() } else { One::one() };
```

Comment:
Now you are pulling this piece of logic here from the internals of `FungiblesAdapter`. Maybe the best would be to just return the actual fee from `correct_and_deposit_fee`? (I get that I proposed part of this and sorry! :see_no_evil:)

In general this `min_coverted_fee` is not a great in general, but that is not your fault :P

Reply:
Yes, I have also thought of this, it makes more sense.